staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -351,22 +351,20 @@ type WaitFunc func(done <-chan struct{}) <-chan struct{}
 // WaitFor continually checks 'fn' as driven by 'wait'.
 //
 // WaitFor gets a channel from 'wait()'', and then invokes 'fn' once for every value
-// placed on the channel and once more when the channel is closed.
+// placed on the channel and once more when the channel is closed. If the channel is closed
+// and 'fn' returns false without error, WaitFor returns ErrWaitTimeout.
 //
 // If 'fn' returns an error the loop ends and that error is returned, and if
 // 'fn' returns true the loop ends and nil is returned.
 //
-// ErrWaitTimeout will be returned if the channel is closed without fn ever
+// ErrWaitTimeout will be returned if the 'done' channel is closed without fn ever
 // returning true.
+//
+// Be careful, when the 'wait' func returns a closed channel and the 'done' is also closed,
+// the behavior of this function depends on the golang 'select'.
```

Comment:
This is a good point! Though I think the following is more clear, what do you think?
```
When the done channel is closed, because the golang `select` statement is "uniform pseudo-random", the `fn` might still run one or multiple time, though eventually `WaitFor` will return.
```

Reply:
OK